src/spark/state.cpp

```
@@ -78,6 +79,45 @@ unsigned char GetNetworkType() {
+
+    std::vector<unsigned char> vch(script.begin() + 2, script.end() - 1);
+
+    try {
+        coinNetwork = addr.fromByteVector(vch);
+    } catch (...) {
+        return false;
+    }
+    return network == coinNetwork;
+}
+
+std::string ToStringSparkAddress(const CScript script) {
```

cppcheck

[error] Signed integer overflow for expression '1<<31'.

(integerOverflow)